scipy/sparse/linalg/tests/test_special_sparse_arrays.py

```diff
@@ -153,7 +156,161 @@ def test_dot(self, grid_shape, bc, dtype):
             y = lap.dot(x.astype(dtype))
             assert x.shape == y.shape
             assert y.dtype == dtype
+            if x.ndim == 2:
+                yy = lap.toarray() @ x.astype(dtype)
+                assert yy.dtype == dtype
+                np.array_equal(y, yy)
 
     def test_boundary_conditions_value_error(self):
         with pytest.raises(ValueError, match="Unknown value 'robin'"):
             LaplacianNd(grid_shape=(6, ), boundary_conditions='robin')
+
+            
+class TestSakurai:
+    """
+    Sakurai tests
+    """
+    ALLDTYPES = [np.int32, np.int64] + REAL_DTYPES + COMPLEX_DTYPES
```

Comment:
I don't like assigning different values to `ALLCAPS` variables - those are usually fixed. Here I'd also want to know why it's _not_ the same as `INT_DTYPES + REAL_DTYPES + COMPLEX_DTYPES`
```suggestion
    # cannot use small integer types (i.e. all of INT_DTYPES) because of $REASON
    tested_types = [np.int32, np.int64] + REAL_DTYPES + COMPLEX_DTYPES
```

Reply:
@h-vetinari I have already forgotten the specifics of my own unit testing related to 
 # type preservation cannot be smaller than the dtype of the `LinearOperator`. 
Let me review it myself again before making the final adjustments. Maybe I could lower the default dtype of Sakurai.